packages/amazonq/src/app/inline/completion.ts

```diff
@@ -237,6 +238,63 @@ export class AmazonQInlineCompletionItemProvider implements InlineCompletionItem
         await vscode.commands.executeCommand(`aws.amazonq.checkInlineSuggestionVisibility`)
     }
 
+    /**
+     * Check if a completion suggestion is currently active/displayed
+     */
+    public async isCompletionActive(): Promise<boolean> {
+        const session = this.sessionManager.getActiveSession()
+        if (session === undefined || !session.displayed || session.suggestions.some((item) => item.isInlineEdit)) {
+            return false
+        }
+
+        // Use VS Code command to check if inline suggestion is actually visible on screen
+        // This command only executes when inlineSuggestionVisible context is true
+        try {
+            await vscode.commands.executeCommand('aws.amazonq.checkInlineSuggestionVisibility')
+            return true
+        } catch {
+            return false
```

Comment:
have you managed to get into this catch block?

Reply:
I've implemented your timestamp-based solution using performance.now() in the checkInlineSuggestionVisibility()
method and checking if it was called within the last 50ms.